src/traces/scattergl/convert.js

```diff
@@ -279,7 +279,8 @@ proto.updateFast = function(options) {
         yy = y[i];
 
         // check for isNaN is faster but doesn't skip over nulls
-        if(!isNumeric(xx) || !isNumeric(yy)) continue;
+        if(!isNumeric(yy)) continue;
+        if(!isNumeric(xx) && !(xx instanceof Date)) continue;
```

Comment:
@etpinard @alexcjohnson I'd just like to clarify something here. Testing with `Lib.isDateTime` is useful here as long as we actually do something with the broader set of options, in particular, string representations of dates. So there's got to be logic to check the type and dispatch to either straight-through processing (no conversion, i.e. epoch milliseconds or `Date` objects that are handled now) or conversion from the string representations of dates to epoch milliseconds or `Date` objects. Should I go ahead and do these things? As mentioned above, I'd keep the fast path fast by testing the first legitimate value only, and if it's suitable for straight-through processing (STP) then the entire array would be assumed to be so.